src/git-command-manager.ts

```diff
@@ -150,22 +155,10 @@ class GitCommandManager {
       args.push(arg)
     }
 
```

Comment:
Refactored most of the lines below into a retry helper. I needed to add retry logic to a new place.